test/config.js

```diff
@@ -193,6 +176,114 @@ test('valid serviceName => active', function (t) {
   t.end()
 })
 
+test('serviceName defaults to package name', function (t) {
+  // TODO:
+  // - Move up top with other requires?
+  // - Switch to some promisified fs module?
+  var mkdirp = util.promisify(require('mkdirp'))
+  var rimraf = util.promisify(require('rimraf'))
+  var writeFile = util.promisify(fs.writeFile)
+  var symlink = util.promisify(fs.symlink)
+  var exec = util.promisify(cp.exec)
+
+  function testServiceConfig (pkg, handle) {
+    var tmp = path.join(os.tmpdir(), 'elastic-apm-node-test')
+    var files = [
+      {
+        action: 'mkdirp',
+        dir: tmp
+      },
+      {
+        action: 'create',
+        path: path.join(tmp, 'package.json'),
+        contents: JSON.stringify(pkg)
+      },
+      {
+        action: 'create',
+        path: path.join(tmp, 'index.js'),
+        contents: `
+          var apm = require('elastic-apm-node').start()
+          console.log(JSON.stringify(apm._conf))
+        `
+      },
+      {
+        action: 'mkdirp',
+        dir: path.join(tmp, 'node_modules')
+      },
+      {
+        action: 'symlink',
+        from: path.resolve(__dirname, '..'),
+        to: path.join(tmp, 'node_modules/elastic-apm-node')
+      }
+    ]
+
+    // NOTE: Reduce the sequence to a promise chain rather
+    // than using Promise.all(), as the tasks are dependent.
+    let promise = files.reduce((p, file) => {
+      return p.then(() => {
+        switch (file.action) {
+          case 'create': {
+            return writeFile(file.path, file.contents)
+          }
+          case 'mkdirp': {
+            return mkdirp(file.dir)
+          }
+          case 'symlink': {
+            return symlink(file.from, file.to)
+          }
+        }
+      })
+    }, Promise.resolve())
+
+    promise = promise
+      .then(() => {
+        return exec('node index.js', {
+          cwd: tmp
+        })
+      })
+      .then(result => {
+        return JSON.parse(result.stdout)
+      })
+
+    return pFinally(promise, () => {
+      return rimraf(tmp)
+    })
+  }
+
+  t.test('should be active when valid', function (t) {
+    var pkg = {
+      name: 'valid'
+    }
+
+    return testServiceConfig(pkg).then(conf => {
+      t.equal(conf.active, true)
```

Comment:
I would check that `serviceName` is correctly set to `valid` as well

Reply:
fixed